src/interfaces/UserStorageKernel.java

```diff
@@ -0,0 +1,58 @@
+package interfaces;
+import components.sequence.Sequence;
+import components.sequence.Sequence1L;
+import components.queue.Queue;
+import components.standard.Standard;
+
+/**
+ * User storage kernel with primary methods.
+ */
+public interface UserStorageKernel extends Standard<UserStorage> {
+    /**
+     * User class for tracking information.
+     */
+    public final class User {
```

Comment:
Look into "java records." This is fine, but you might get more value out of breaking the discipline a bit.